service/internal/config/config.go

```diff
@@ -89,7 +91,15 @@ func LoadConfig(key string, file string) (*Config, error) {
 }
 
 func (c *Config) LogValue() slog.Value {
-	redactedConfig := util.RedactSensitiveData(c)
+	var sensitiveKeys = c.SensitiveKeys
+
+	redactedInterface := util.RedactSensitiveData(c, sensitiveKeys)
+	redactedConfig, ok := redactedInterface.(*Config) // Direct type assertion
+	if !ok {
+		log.Println("Error asserting redacted configuration to *Config type")
```

Comment:
We prefer`slog` to `log`, but given you are already processing a logline at this point maybe you can just add the error to the output? maybe 

```
  values = append(values, slog.String("logError", "unable to redact config"))
```
or similar